nemo/collections/llm/modelopt/distill/utils.py

```diff
@@ -114,7 +113,9 @@ def adjust_distillation_model_for_mcore(
     """Extra modifications to ``mtd.DistillationModel`` required for Megatron-Core."""
     # Get rid of ModelOpt Distillation state
     # NOTE: If re-placed, above losses need modifcation as `TransformerConfig` has non-pickleable elements.
-    mto.ModeloptStateManager(model)._state.pop()
+    existing_state = mto.ModeloptStateManager(model).state_dict()
+    assert len(existing_state) == 1 and existing_state[0][0] == "kd_loss", f"{existing_state=}"
+    mto.ModeloptStateManager.transfer_state_dict(model, torch.nn.Module())
```

Comment:
This is a small bug fix: Previously modelopt state for distillation was not removed but just set to empty hence we cannot again load ptq ckpt for student.

This fix now allows us to load student as a quantized ckpt hence allow QAT with distillation